src/vm/database/clarity_db.rs

```diff
@@ -680,6 +716,41 @@ impl<'a> ClarityDatabase<'a> {
             .expect("Failed to get block data.")
     }
 
+    /// Gets the `StacksBlockId` for the *current* chain tip Stacks block height, converts this
+    /// `StacksBlockId` to a `SortitionId` and then gets the `BurnchainHeaderHash` from the
+    /// `BurnStateDB` using this `SortitionId`.
+    ///
+    /// Returns Some() if `0 <= height < current_burnblock_height`, and None otherwise.
+    pub fn get_burnchain_block_header_hash_for_burnchain_height(
+        &mut self,
+        burnchain_block_height: u32,
+    ) -> Option<BurnchainHeaderHash> {
+        let current_stacks_height = self.get_current_block_height();
+        let id_bhh = self.get_index_block_header_hash(current_stacks_height);
```

Comment:
You should use the _parent_ ID block hash, rather than the current stacks height. The reason for that is that the current block hash is evaluated with a placeholder by the miner, so its consensus hash, etc. isn't contained in the headers db yet (which only contains information about already-mined blocks).

Reply:
Per my other comment, this would be the same as the value of `current` passed into `ClarityInstance::begin_block()`.